src/pip/_internal/wheel_builder.py

```diff
@@ -87,6 +79,14 @@ def _should_build(
         )
         return False
 
+    if not req.use_pep517 and not is_wheel_installed():
+        # we don't build legacy requirements if wheel is not installed
+        logger.info(
+            "Using legacy setup.py install for %s, "
+            "since package 'wheel' is not installed.", req.name,
```

Comment:
I also see that there are no unit test which check for this log message. Perhaps this PR is a good opportunity to add one?

Reply:
I don't think it's worth the trouble, because #8102 will probably cause some rework in this area.

Reply:
In that case we don't need a test.